### PR TITLE
Update reference to Python web app quickstart

### PR DESCRIPTION
**Pull request by Dickson-Mwendia**
Fixes a couple of broken links, update from https://learn.microsoft.com/en-us/azure/active-directory/develop/quickstart-v2-python-webapp to  https://learn.microsoft.com/en-us/azure/active-directory/develop/web-app-quickstart?pivots=devlang-python

### CODE REVIEW

**Comment by Dickson-Mwendia** [comment]
#assign-reviewer: @rayluo

**Comment by rayluo** [comment]
Thanks for the catch.

The [old link](https://learn.microsoft.com/en-us/azure/active-directory/develop/quickstart-v2-python-webapp) currently shows "... This probably isn't the page you were expecting. *While we work on a fix* ...". Are we actually working on a fix to restore the old link, or are we permanently moving to the new link? In other words, what will be the canonical link going forward?

**Comment by Dickson-Mwendia** [comment]
We're moving to the new link that holds all quickstarts for a particular app type.  Soon we'll remove all references to the old quickstarts and delete them. However, we'll also apply redirects to ensure users do not land on already deleted pages.

**Review by rayluo — APPROVED** [review]
Thanks for the catch and clarifying that the new link is the new canonical link.

File: docs/index.rst [hunk 1]
```diff
@@ -24,15 +24,15 @@ MSAL Python supports some of them.
...
 
-    <!-- Original diagram came from https://docs.microsoft.com/en-us/azure/active-directory/develop/media/scenarios/scenarios-with-users.svg -->
+    <!-- Original diagram came from https://docs.microsoft.com/azure/active-directory/develop/media/scenarios/scenarios-with-users.svg -->
     <!-- Don't know how to include images into Sphinx, so we host it from github repo instead -->
     <img src="https://raw.githubusercontent.com/AzureAD/microsoft-authentication-library-for-python/dev/docs/scenarios-with-users.svg"
         usemap="#public-map"><!-- Derived from http://www.image-map.net/ but we had to manually add unique map id -->
     <map name="public-map">
         <area target="_blank" coords="110,150,59,94" shape="rect"
-            alt="Web app" title="Web app" href="https://docs.microsoft.com/en-us/azure/active-directory/develop/quickstart-v2-python-webapp">
+            alt="Web app" title="Web app" href="https://learn.microsoft.com/azure/active-directory/develop/web-app-quickstart?pivots=devlang-python>
```
Comment (rayluo) [inline docs/index.rst:33]: It turns out, this line and the next line were both missing a trailing quotation mark.

Now I fixed it in my latest commit.

CC: @Dickson-Mwendia

**Pull request merged**